index.js

```diff
@@ -84,10 +85,47 @@ Lexer.prototype = {
    * @api private
    */
 
-  tok: function(type, val){
-    return val === undefined ?
-      {type: type, line: this.lineno} :
-      {type: type, line: this.lineno, val: val};
+  tok: function(type, val, columnIncrement){
+    var res = {type: type, line: this.lineno};
+
+    if (this.colno !== null) res.col = this.colno;
+    if (val !== undefined) res.val = val;
+
+    if (columnIncrement !== undefined) this.incrementColumn(columnIncrement);
```

Comment:
It might be safer to assert that `columnIncrement` is a number, then explicitly pass `0` if the column is not being incremented.